e2e/src/test/java/no/sysco/testing/kafka/e2e/ContainersIT.java

```diff
@@ -0,0 +1,194 @@
+                                .extract()
+                                .as(MessageJsonRepresentation[].class));
+        assertTrue(messageJsonRepresentations.size() > 0);
+    }
+
+    @Test
+    public void test_data_pipeline_flow_successful() {
+        String id = UUID.randomUUID().toString();
+        String from = UUID.randomUUID().toString();
+        String to = UUID.randomUUID().toString();
+        String text = UUID.randomUUID().toString();
+
+        MessageJsonRepresentation messageJsonRepresentation =
+                new MessageJsonRepresentation(id, from, to, text);
+
+        RestAssured.given()
+                .contentType(ContentType.JSON)
+                .body(messageJsonRepresentation)
+                .post(HTTP_PRODUCER_EXPOSED + "/messages")
+                .then()
+                .statusCode(202);
+
+        await()
+                .atMost(70, TimeUnit.SECONDS)
+                .untilAsserted(
+                        () -> {
+                            MessageJsonRepresentation jsonPresentation =
+                                    RestAssured.given()
+                                            .get(JSON_SERVER_EXPOSED + "/messages/" + id)
+                                            .then()
+                                            // .statusCode(200)
+                                            .extract()
+                                            .as(MessageJsonRepresentation.class);
+
+                            assertNotNull(jsonPresentation);
+                        });
+    }
+
+    private static void setZookeeperAndKafka() {
+        kafka = new KafkaContainer(CONFLUENT_PLATFORM_VERSION);
+        kafka.start();
+    }
+
+    private static void setSchemaRegistry() {
+        // get network alias to be able connect other container to env
+        KAFKA_BROKER_INSIDE_DOCKER_ENV = "PLAINTEXT://" + kafka.getNetworkAliases().get(0) + ":9092";
+        schemaRegistry =
+                new GenericContainer("confluentinc/cp-schema-registry:" + CONFLUENT_PLATFORM_VERSION)
+                        .withExposedPorts(8081)
+                        .withNetwork(kafka.getNetwork())
+                        .withEnv("SCHEMA_REGISTRY_HOST_NAME", "localhost")
+                        .withEnv("SCHEMA_REGISTRY_LISTENERS", "http://0.0.0.0:8081")
+                        .withEnv("SCHEMA_REGISTRY_KAFKASTORE_BOOTSTRAP_SERVERS", KAFKA_BROKER_INSIDE_DOCKER_ENV)
+                        .waitingFor(Wait.forHttp("/subjects").forStatusCode(200));
+        schemaRegistry.start();
+        SCHEMA_REGISTRY_INSIDE_DOCKER_ENV =
+                "http://" + schemaRegistry.getNetworkAliases().get(0) + ":8081";
+    }
+
+    private static void setJsonServer() {
+        jsonServer =
+                new GenericContainer("zhenik/json-server")
+                        .withExposedPorts(80)
+                        // all containers put in same network
+                        .withNetwork(kafka.getNetwork())
```

Comment:
I would highly recommend declaring a network explicitly (via `Network.newNetwork()`) instead of using Kafka's network.
In fact, after https://github.com/testcontainers/testcontainers-java/pull/1458 we may stop setting the network by default on Kafka's container

Reply:
done